content/blog/2024/hackbots-part-2/index.md

```diff
@@ -0,0 +1,188 @@
+---
+title: "The Hackbots Project II"
+subtitle: "Making Simple Robots That Anyone Can Build (Pt. 2)"
+date: 2024-03-18T07:40:38Z
+tags:
+    - Rasberry Pi Pico
+    - Robots
+    - Electronics
+draft: false
+author: Paul Williams
+author_email: phyushin@gmail.com
+listing_image: hackbot.jpg
+---
+
+# Continuing the Bot Building Process
+So, we'll continue from where we left on last a little while ago with our bot building blog [posts][1]. 
+We'd got to the MK IV if memory serves... Next on the list of things to do is to make our bot interface page nice and pretty.
+
+## Enter, CSS
+Because we're using a very simple webserver on the [Raspbery Pi Pico W][2] and because the code handily returned from our `generateHTML()` function is effectively the page that's served by the websever; we know this is a good place to add a little customisation for our robot - make our page a little _prettier_.
+
+## CodePen
+When prototyping, it's always much more satisfying to see the changes happen quickly, as opposed to editing our `generateHTML()` function, uploading to the bot, connecting to our bot, visiting the url... it's a lot of steps - so, to make this a bit easier we're going to use a nice tool called [CodePen][3], our first step is to rip out the HTML from the `generateHTML()` function and paste it into the HTML window, like so:
+![HTML & CSS editor windows above, results below](./images/codepen_screenshot.png)
```

Comment:
Needs replacing with a `image` shortcode tag, while the images don't need anything special it'll allow Hugo to handle resizing/rendering of the images to efficient formats rather than using the raw image.

```
{{< image src="images/codepen_screenshot.png" title="HTML & CSS editor windows above, results below">}}
```